### PR TITLE
tkt-41365: feat(middlewared): limit sentry crash reporting

### PR DESCRIPTION
**Pull request by william-gr**
Ticket:	#41365

### CODE REVIEW

**Comment by william-gr** [comment]
That would be yet another dependency which would consume time for us to port and test, I think its safer for now if we just use base raven package.

File: src/middlewared/middlewared/main.py [hunk 1]
```diff
@@ -183,11 +183,17 @@ def send_error(self, message, errno, reason=None, exc_info=None, etype=None, ext
...
+            elif self.middleware.crash_reporting_semaphore.locked():
+                self.logger.debug('[Crash Reporting] skipped due too many running instances')
             else:
-                extra_log_files = (('/var/log/middlewared.log', 'middlewared_log'),)
-                asyncio.ensure_future(self.middleware.run_in_thread(
-                    self.middleware.crash_reporting.report, sys.exc_info(), None, extra_log_files
-                ))
+                async with self.middleware.crash_reporting_semaphore:
+                    extra_log_files = (('/var/log/middlewared.log', 'middlewared_log'),)
+                    await self.middleware.run_in_io_thread(
```
Comment (themylogin) [inline src/middlewared/middlewared/main.py]: IMO `await` is not appropriate here. I think we can put all this new code (checking if locked, acquire, sending crash report) into a coroutine and just ensure_future here

**Comment by william-gr** [comment]
wait, that wont work = P

**Comment by william-gr** [comment]
@themylogin how is that now?

File: src/middlewared/middlewared/service.py [hunk 2]
```diff
@@ -611,3 +611,6 @@ def sleep_fn():
             job.set_progress(current_progress)
 
         return statuses
+
+    def test(self):
```
Comment (themylogin) [inline src/middlewared/middlewared/service.py]: Probably should be removed

**Pull request merged**